cmake.deps/cmake/TreesitterParserCMakeListsCXX.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.10)
```

Comment:
This entire file doesn't seem to be even used. Is there a reason for this?

Reply:
Not yet; but if we include other parsers that use a `scanner.cc` instead of a `scanner.c`, we will?

Reply:
Should be reflected in the name, though -- maybe `TreeSitterParserCMakeListsScannerC` and (possibly later) `TreeSitterParserCMakeListsScannerCXX`?

Reply:
@dundargoc Is there any reasonable way to combine these three files into a single one? We have three situations, depending on the specific parser:
1. use a C compiler to compile `src/parser.c`
2. use a C compiler to compile `src/parser.c` and `src/scanner.c`
3. use a C++ compiler to compile `src/parser.c` and `src/scanner.cc`

(Current plan -- C, Lua, Vimscript -- only needs options 1 and 2.)

Reply:
1 and 2 can be combined for sure; my [comment](https://github.com/neovim/neovim/pull/15391#issuecomment-1219515823) outlines how.

The third one should be doable but I'd need to test on a c++ parser to be sure.

Reply:
I'm a bit wary that some parsers include extra files in `src`, but we can cross that bridge when we come to it.

https://github.com/tree-sitter/tree-sitter-python uses a `scanner.cc`, for example.